pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -280,7 +280,18 @@ func (ds *Plugin) FetchAttestedNode(ctx context.Context, spiffeID string) (attes
 }
 
 // CountAttestedNodes counts all attested nodes
-func (ds *Plugin) CountAttestedNodes(ctx context.Context) (count int32, err error) {
+func (ds *Plugin) CountAttestedNodes(ctx context.Context, req *datastore.CountAttestedNodesRequest) (count int32, err error) {
+	if countAttestedNodesHasFilters(req) {
+		resp, err := listAttestedNodes(ctx, ds.db, ds.log, &datastore.ListAttestedNodesRequest{
+			ByAttestationType: req.ByAttestationType,
+			ByBanned:          req.ByBanned,
+			ByExpiresBefore:   req.ByExpiresBefore,
+			BySelectorMatch:   req.BySelectorMatch,
+			FetchSelectors:    req.FetchSelectors,
+			ByCanReattest:     req.ByCanReattest,
+		})
+		return int32(len(resp.Nodes)), err
```

Comment:
Discussed this with @FedeNQ offline. Since the changes to avoid retrieving all the records are too complex, I suggest to keep it as is so we minimize the changes required in the datastore, which has a complex code already.